### PR TITLE
Enables production tests on local environments

### PR DESCRIPTION
**Pull request by ajsantander**
Allows starting a local chain with
`npx buidler node`

And running production tests against it:
`npm run test:prod`

The exact details of how to run the tests in each case can be found in the 2 new ci jobs that come in the PR:
`job-production-tests-local` and `job-production-tests-local-ovm`

The OVM tests compile regular EVM artifacts, but deploy an OVM configuration in the local chain. Note that this does not test the OVM bridge, simply that the deployed instance can perform basic production functionalities like transfer, exchange, etc.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -354,8 +354,12 @@ const getUsers = ({ network = 'mainnet', user, useOvm = false } = {}) => {
 		ropsten: Object.assign({}, base),
 		goerli: Object.assign({}, base),
 		'goerli-ovm': Object.assign({}, base),
+		local: Object.assign({}, base),
```
Comment (justinjmoses) [inline index.js:357]:
```suggestion
		local: Object.assign({}, base, { owner: '0x...' }),
```

File: test/prod/EtherCollateral.prod.js [hunk 2]
```diff
@@ -1,34 +1,50 @@
...
-	let network;
+	let network, deploymentPath;
 
-	let EtherCollateral, AddressResolver;
-	let SynthsETH;
+	let EtherCollateral, AddressResolver, Depot;
+	let SynthsETH, SynthsUSD;
 
-	before('prepare', async () => {
+	before('prepare', async function() {
```
Comment (justinjmoses) [inline test/prod/EtherCollateral.prod.js:27]: why did you remove fat arrow? To keep `this`? It should work as expected to nest fat arrows from `contract(..)` down and preserve context, no?

**Review by justinjmoses — APPROVED** [review]
Just a couple of comments

**Review by i-stam — APPROVED** [review]
Thanks for `bootstrapLocal.js`.  I 'm already using it, very helpful!

File: test/prod/EtherCollateral.prod.js (hunk 2 above)
Comment (ajsantander) [reply test/prod/EtherCollateral.prod.js:27]: This was necessary to be able to call this.skip() and jump out of a describe dynamically in mocha.

File: index.js (hunk 1 above)
Comment (ajsantander) [reply index.js:357]: Sorry, forgot to include this and merged. Will open a new PR with this suggestion.
